packages/st2/Makefile

```
@@ -1,34 +1,55 @@
 WHEELDIR ?= /tmp/wheelhouse
 ST2_COMPONENT := $(notdir $(CURDIR))
 ST2PKG_RELEASE ?= 1
-ST2PKG_VERSION ?= $(shell python -c "from $(ST2_COMPONENT) import __version__; print __version__,")
 CHANGELOG_COMMENT ?= "automated build, version: $(ST2PKG_VERSION)"
 
 ifneq (,$(wildcard /etc/debian_version))
 	DEBIAN := 1
 	#DEB_EPOCH := $(shell echo $(ST2PKG_VERSION) | grep -q dev || echo '1')
 	DEB_DISTRO := $(shell lsb_release -cs)
+else ifneq (,$(wildcard /etc/centos-release))
+	EL_DISTRO:=centos
+	EL_VERSION := $(shell cat /etc/centos-release | grep -oP '(?<= )[0-9]+(?=\.)')
+	REDHAT := 1
+else ifneq (,$(wildcard /etc/redhat-release))
+	EL_DISTRO:=redhat
+	EL_VERSION := $(shell cat /etc/centos-release | grep -oP '(?<= )[0-9]+(?=\.)')
+	REDHAT := 1
 else
 	REDHAT := 1
 	DEB_DISTRO := unstable
 endif
 
-ifneq (,$(wildcard /usr/share/python/st2python/bin/python))
+ifeq ($(DEB_DISTRO),bionic)
+	PYTHON_BINARY := /usr/bin/python3
+	PIP_BINARY := pip
+else ifeq ($(EL_VERSION),8)
+	PYTHON_BINARY := /usr/bin/python3
+	PIP_BINARY := /usr/local/bin/pip3
+	PYTHON_ALT_BINARY := python3
+else ifneq (,$(wildcard /usr/share/python/st2python/bin/python))
 	PATH := /usr/share/python/st2python/bin:$(PATH)
 	PYTHON_BINARY := /usr/share/python/st2python/bin/python
-else ifeq ($(DEB_DISTRO),bionic)
-	PYTHON_BINARY := /usr/bin/python3
+	PIP_BINARY := pip
+	PYTHON_ALT_BINARY := python
 else
 	PYTHON_BINARY := python
+	PIP_BINARY := pip
+	PYTHON_ALT_BINARY := python
 endif
 
-# Don't include Mistral runner on Bionic
+# Don't include Mistral runner on Bionic or EL8
 ifeq ($(DEB_DISTRO),bionic)
 	RUNNERS := $(shell ls -I mistral* ../contrib/runners)
+else ifeq ($(EL_VERSION),8)
+	RUNNERS := $(shell ls -I mistral* ../contrib/runners)
 else
 	RUNNERS := $(shell ls ../contrib/runners)
 endif
 
+# Moved from top of file to handle when only py2 or py3 available
+ST2PKG_VERSION ?= $(shell $(PYTHON_BINARY) python3 -c "from $(ST2_COMPONENT) import __version__; print(__version__),")
```

Is the expected `shell $(PYTHON_BINARY) python3` part of the cmd ?

Shouldn't it be 
```suggestion
ST2PKG_VERSION ?= $(shell $(PYTHON_BINARY) -c "from $(ST2_COMPONENT) import __version__; print(__version__),")
```
?

---

yes, good catch.